geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/ProtobufSimpleAuthenticator.java

```diff
@@ -40,20 +42,28 @@ public void receiveMessage(InputStream inputStream, OutputStream outputStream,
     properties.setProperty("username", authenticationRequest.getUsername());
     properties.setProperty("password", authenticationRequest.getPassword());
 
+    authorizer = null; // authenticating a new user clears current authorizer
     try {
       Object principal = securityManager.authenticate(properties);
-      authenticated = principal != null;
+      if (principal != null) {
+        authorizer = new ProtobufSimpleAuthorizer(principal, securityManager);
+      }
     } catch (AuthenticationFailedException e) {
-      authenticated = false;
+      authorizer = null;
     }
 
-    AuthenticationAPI.SimpleAuthenticationResponse.newBuilder().setAuthenticated(authenticated)
+    AuthenticationAPI.SimpleAuthenticationResponse.newBuilder().setAuthenticated(isAuthenticated())
         .build().writeDelimitedTo(outputStream);
   }
 
   @Override
   public boolean isAuthenticated() {
-    return authenticated;
+    return authorizer != null;
```

Comment:
I must disagree with this logic. Something is NOT `authenticated` just because there is an authorizer populated. A authorizer should be constructed BECAUSE something was `authenticated`